src/lower/merge_lattice.cpp

```diff
@@ -1119,8 +1207,18 @@ ostream& operator<<(ostream& os, const MergeLattice& ml) {
 }
 
 bool operator==(const MergeLattice& a, const MergeLattice& b) {
-  auto& apoints = a.points();
-  auto& bpoints = b.points();
+  auto apoints = a.points();
+  auto bpoints = b.points();
+  struct pointSort {
+    bool operator()(const MergePoint& a, const MergePoint& b) {
```

Comment:
Same question about sorting. I think the MergeLattice's points should always be sorted so it seems strange that we need to sort before a comparison

Reply:
See above. You sorted before in the original `unionLattices(...)` anyways when taking the cartesian product so I just took the code from there.

Reply:
I see, I was just wondering why a sort inside one of the MergeLattice operators since the points should always be sorted. But as above, I also ran into the issue of points on the same level being permuted. 

I think this is fine.